postgres.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type Database struct {
+	Pool *pgxpool.Pool
+}
+
+func InitPostgresDatabase(databaseURL string) (*Database, error) {
+	config, err := pgxpool.ParseConfig(databaseURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse database URL: %w", err)
+	}
+
+	pool, err := pgxpool.NewWithConfig(context.Background(), config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create connection pool: %w", err)
+	}
+
+	if err := pool.Ping(context.Background()); err != nil {
+		return nil, fmt.Errorf("failed to ping database: %w", err)
+	}
+
+	return &Database{Pool: pool}, nil
+}
+
+func (db *Database) Close() {
+	db.Pool.Close()
+}
+
+type Fee struct {
+	PartnerID     int
+	FeePercentage float64
+}
+
+type FeeModel struct {
+	DB *pgxpool.Pool
+}
+
+func NewFeeModel(db *pgxpool.Pool) *FeeModel {
+	return &FeeModel{DB: db}
+}
+
+func (m *FeeModel) GetByPartnerApiKey(api_key string) (*Fee, error) {
```

Comment:
Perhaps adding a default fee as a parameter would be a good idea. If there is now row containing the api key, the default fees will  be returned. We have to check what we do for the id.